proxy/src/telemetry/metrics/latency.rs

```diff
@@ -0,0 +1,267 @@
+#![deny(missing_docs)]
+use control::pb::telemetry::LatencyBucket;
+
+use std::{ops, slice, u32};
+use std::default::Default;
+use std::time::Duration;
+
+/// The number of buckets in a  latency histogram.
```

Comment:
There's this comment in `Histogram::new`:

```
            // TODO: it would be nice if we didn't have to hard-code each
            //       individual bucket and could use Rust ranges or something.
            //       However, because we're using a raw fixed size array rather
            //       than a vector (as we don't ever expect to grow this array
            //       and thus don't _need_ a vector) we can't concatenate it
            //       from smaller arrays, making it difficult to construct
            //       programmatically...
```

Reply:
I thought that, given that the number of buckets _is_ fixed, using a fixed-size array was appropriate, even though it requires a hard-coded constant for the number of buckets. 

Our other options include using a `Vec<Bucket>`, which seems unnecessary as we never expect to grow the list of buckets, and  `Box<[Bucket]>` which isn't growable but still heap-allocates. We could also use a `&'a [Bucket]`, but that would A. introduce a lifetime bound on the type, and B) still require a `Vec<Bucket>`, `Box<[Bucket]`, or `[Bucket; N]` to be created _somewhere else_; it just moves the responsibility for creating the actual array upstream.